php-classes/Slate/CBL/CompetenciesRequestHandler.php

```diff
@@ -171,13 +171,21 @@ public static function handleDemonstrationsRequest(Competency $Competency)
 
         // query demonstrations sums
         try {
-            $skillDemonstrations = DB::allRecords(
-                'SELECT Demonstration.ID AS DemonstrationID, Demonstration.StudentID, DemonstrationSkill.SkillID, DemonstrationSkill.Level'
-                .' FROM (SELECT ID FROM `%s` WHERE CompetencyID = %u) AS Skill'
-                .' JOIN `%s` DemonstrationSkill'
-                .'  ON DemonstrationSkill.SkillID = Skill.ID'
-                .' JOIN (SELECT ID, StudentID FROM `%s` WHERE StudentID IN (%s)) Demonstration'
-                .'  ON Demonstration.ID = DemonstrationSkill.DemonstrationID'
+            $skillDemonstrations = DB::allRecords('
```

Comment:
NOWDOC doesn't support variables. HEREDOC does, but both will throw the indentation outside of the PHP. I think that concatenating the strings is prone to error. I don't think all of the extra whitespace causes much of an issue.